specification/mariadb/resource-manager/Microsoft.DBforMariaDB/preview/2020-01-01-privatepreview/DataEncryptionKeys.json

```diff
@@ -0,0 +1,395 @@
+        "parameters": [
+          {
+            "$ref": "#/parameters/ResourceGroupParameter"
+          },
+          {
+            "$ref": "#/parameters/ServerNameParameter"
+          },
+          {
+            "name": "keyName",
+            "in": "path",
+            "description": "The name of the MariaDB Server key to be deleted.",
+            "required": true,
+            "type": "string"
+          },
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "Successfully deleted the MariaDB Server key."
+          },
+          "202": {
+            "description": "Accepted"
+          },
+          "204": {
+            "description": "The specified MariaDB Server key does not exist."
+          },
+          "default": {
+            "description": "Error response describing why the operation failed.",
+            "schema": {
+              "$ref": "#/definitions/CloudError"
+            }
+          }
+        },
+        "x-ms-long-running-operation": true,
+        "x-ms-examples": {
+          "Delete the MariaDB Server key": {
+            "$ref": "./examples/ServerKeyDelete.json"
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "ProxyResource": {
+      "description": "Resource properties.",
+      "properties": {
+        "id": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource ID"
+        },
+        "name": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource name."
+        },
+        "type": {
+          "readOnly": true,
+          "type": "string",
+          "description": "Resource type."
+        }
+      },
+      "x-ms-azure-resource": true
+    },
+    "ServerKeyListResult": {
+      "description": "A list of MariaDB Server keys.",
+      "type": "object",
+      "properties": {
+        "value": {
+          "description": "Array of results.",
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/ServerKey"
+          },
+          "readOnly": true
+        },
+        "nextLink": {
+          "description": "Link to retrieve next page of results.",
+          "type": "string",
+          "readOnly": true
+        }
+      }
+    },
+    "ServerKeyProperties": {
+      "description": "Properties for a key execution.",
+      "required": [
+        "serverKeyType"
+      ],
+      "type": "object",
+      "properties": {
+        "serverKeyType": {
+          "description": "The key type like 'ServiceManaged', 'AzureKeyVault'.",
+          "enum": [
+            "AzureKeyVault"
+          ],
+          "type": "string",
+          "x-ms-enum": {
+            "name": "ServerKeyType",
+            "modelAsString": true
+          }
+        },
+        "uri": {
```

Comment:
What is this uri? Can it be anything or only a keyvault key?

How do you access keyvault? Do you plan to use MSI to access this key?

Reply:
looking at your RP manifest change, you'll expose MSI on the /server resource. You should document this in swagger too.

Reply:
This Uri is Azure key vault.  Yes.  We are using MSI to access this key

Reply:
> looking at your RP manifest change, you'll expose MSI on the /server resource. You should document this in swagger too.

Done

Reply:
This Uri is Azure key vault. Yes. We are using MSI to access this key. Included the resource identity as well in the manifest

Reply:
You should call this property (and the other 'uri' one) something more specific. Are you planning to add a LinkedAccessCheck to this? If so I'd make it specific to the purpose of keyvault resource id. When you introduce more serverKeyTypes, you can use discriminators to change the schema to be specific to each type introduced.

Reply:
As discussed over the call, we will address these before public preview